lib/rules/prefer-template.js

```diff
@@ -39,33 +39,25 @@ function getTopConcatBinaryExpression(node) {
 }
 
 /**
- * Determines whether a given node is a octal escape sequence
+ * Checks whether or not a node contains a string literal with an octal or non-octal decimal escape sequence
  * @param {ASTNode} node A node to check
- * @returns {boolean} `true` if the node is an octal escape sequence
+ * @returns {boolean} `true` if at least one string literal within the node contains
+ * an octal or non-octal decimal escape sequence
  */
-function isOctalEscapeSequence(node) {
-
-    // No need to check TemplateLiterals – would throw error with octal escape
-    const isStringLiteral = node.type === "Literal" && typeof node.value === "string";
-
-    if (!isStringLiteral) {
-        return false;
+function hasOctalOrNonOctalDecimalEscapeSequence(node) {
```

Comment:
I like how you simplified these two helpers down to a single recursive function. It's easier to follow.